GoogleDataTransport/GDTCORLibrary/GDTCORUploadCoordinator.m

```diff
@@ -163,23 +163,27 @@ + (BOOL)supportsSecureCoding {
 
 - (instancetype)initWithCoder:(NSCoder *)aDecoder {
   GDTCORUploadCoordinator *sharedCoordinator = [GDTCORUploadCoordinator sharedInstance];
-  @try {
-    sharedCoordinator->_targetToInFlightPackages =
-        [aDecoder decodeObjectOfClass:[NSMutableDictionary class]
-                               forKey:ktargetToInFlightPackagesKey];
+  dispatch_sync(sharedCoordinator->_coordinationQueue, ^{
```

Comment:
I remember we had a case when using  `dispatch_sync` in coder methods led to a deadlock. It looks safe here, but just would like to double check if you can see a case when the deadlock may occur?

Reply:
I looked in the history and didn't think this was one of the affected places. I tested using test apps and it generally looked ok.